### DISCUSSION TITLE
How do I access the bindingContext constructor?

### DISCUSSION THREAD

**Discussion by tscpp** [knockout/tko#149] [Q&A] [answered]
I am making my own component registration to use shadow-dom in my components. However I can't make a new binding context because the "bindingContext" constructor is not exported in the build. I can't simply copy and paste the constructor because the constructor [utilizes symbols](https://github.com/knockout/tko/blob/afaa804bed944508d469c99fe69e32e62c4921b4/packages/bind/src/bindingContext.ts#L68).

https://github.com/knockout/tko/blob/afaa804bed944508d469c99fe69e32e62c4921b4/packages/bind/src/bindingContext.ts#L24

----

I solved the specific problem I had by using the "extendContextCallback" property in the applyBindings function which was new to me.

**Comment by brianmhunt** [comment, marked as answer]
Thanks @tscpp .

Indeed it isn't exported.  I think it should be, too.

One option is to make a custom build, then you can import it like this (for both the build and your own purposes):

```
import {
  bindingContext as BindingContextConstructor
} from "@tko/bind";
```

Another option is if you have another `$context` you can use `createChildContext`.

Both feel like workarounds, but I hope this is helpful food for thought.
